client/src/main/java/io/pravega/client/control/impl/ModelHelper.java

```diff
@@ -271,15 +271,25 @@ public static ReaderGroupConfig encode(Controller.ReaderGroupConfiguration rgCon
                 .generation(rgConfig.getGeneration())
                 .readerGroupId(UUID.fromString(rgConfig.getReaderGroupId()))
                 .startingStreamCuts(rgConfig.getStartingStreamCutsList().stream()
-                                    .collect(Collectors.toMap(streamCut -> Stream.of(streamCut.getStreamInfo().getScope(),
-                                    streamCut.getStreamInfo().getStream()),
-                                    streamCut -> new StreamCutImpl(Stream.of(streamCut.getStreamInfo().getScope(),
-                                    streamCut.getStreamInfo().getStream()), getSegmentOffsetMap(streamCut.getStreamInfo().getScope(), streamCut.getStreamInfo().getStream(), streamCut.getCutMap())))))
+                        .collect(Collectors.toMap(streamCut -> Stream.of(streamCut.getStreamInfo().getScope(),
+                                streamCut.getStreamInfo().getStream()),
+                                streamCut -> {
```

Comment:
It would be good to encapsulate this lambda in a method and invoke that method here, so we can reuse code instead of repeat, same method can also be invoked in CreateReaderGroupTask.

Reply:
The lambda has been encapsulated into a method which is used in both classes.